fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/provider/impl/S3Provider.java

```diff
@@ -0,0 +1,441 @@
+                if (transientUserData.getUploadIdForFailureOperationOutcomes() == null) {
+                    transientUserData.setUploadIdForFailureOperationOutcomes(BulkDataUtils.startPartUpload(client, bucketName, transientUserData.getUniqueIDForImportFailureOperationOutcomes(), true));
+                }
+
+                transientUserData.getDataPacksForFailureOperationOutcomes().add(BulkDataUtils.multiPartUpload(client, bucketName, transientUserData.getUniqueIDForImportFailureOperationOutcomes(), transientUserData.getUploadIdForFailureOperationOutcomes(), new ByteArrayInputStream(transientUserData.getBufferStreamForImportError().toByteArray()), transientUserData.getBufferStreamForImportError().size(), (int) transientUserData.getPartNumForFailureOperationOutcomes()));
+                if (logger.isLoggable(Level.FINE)) {
+                    logger.fine("pushImportOperationOutcomesToOS: " + transientUserData.getBufferStreamForImportError().size()
+                            + " bytes were successfully appended to COS object - " + transientUserData.getUniqueIDForImportFailureOperationOutcomes());
+                }
+                transientUserData.setPartNumForFailureOperationOutcomes(transientUserData.getPartNumForFailureOperationOutcomes() + 1);
+                transientUserData.getBufferStreamForImportError().reset();
+            }
+        } catch (Exception e) {
+            throw new FHIRException("Unable to write to S3 OperationOutcome File", e);
+        }
+    }
+
+    @Override
+    public void createSource() throws FHIRException {
+        if (client != null) {
+            if (create && !this.exists()) {
+                CreateBucketRequest req = new CreateBucketRequest(bucketName);
+                client.createBucket(req);
+            }
+        } else {
+            throw new FHIRException("Client is not created");
+        }
+    }
+
+    public AmazonS3 getClient() {
+        return client;
+    }
+
+    public SparkParquetWriter getParquetWriter() {
+        return parquetWriter;
+    }
+
+    @Override
+    public void close() throws Exception {
+        logger.fine("closing the S3Wrapper");
+        if (parquetWriter != null) {
+            parquetWriter.close();
+        }
+    }
+
+    @Override
+    public void writeResources(String mediaType, List<ReadResultDTO> dtos) throws Exception {
+        for (ReadResultDTO dto : dtos) {
+            switch (mediaType) {
+            case FHIRMediaType.APPLICATION_PARQUET:
+                if (chunkData.getPageNum() > chunkData.getLastPageNum() || chunkData.isFinishCurrentUpload()) {
+                    pushFhirParquetToCos(dto.getResources());
+                    chunkData.setLastWritePageNum(chunkData.getPageNum());
+                }
+                break;
+            case FHIRMediaType.APPLICATION_NDJSON:
+            default:
+                if (chunkData.getPageNum() > chunkData.getLastPageNum() || chunkData.isFinishCurrentUpload()) {
```

Comment:
Looks like we accidentally lost our "do we have enough bytes to start a partUpload" check from here.
I believe what we wanted to do is remove the `chunkData.getPageNum() > chunkData.getLastPageNum()` check from here, but *leave* the `chunkData.getBufferStream().size() > COS_PART_MINIMALSIZE` one
(see https://github.com/IBM/FHIR/pull/1999/commits/61e8e40f7944aff4d8083f7e89e7b2c23b54e2e2#diff-83b573e778b2ec67186ea08aa4515c6d7e6b3583c592abf3fc44cbb3240a7f06L343).